includes/class-wc-payments-upe-checkout.php

```diff
@@ -0,0 +1,299 @@
+<?php
+/**
+ * Class WC_Payments_Checkout
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay;
+
+use Exception;
+use WC_Payments;
+use WC_Payments_Account;
+use WC_Payments_Customer_Service;
+use WC_Payments_Utils;
+use WCPay\Fraud_Prevention\Fraud_Prevention_Service;
+use WCPay\Payment_Methods\UPE_Payment_Gateway;
+use WCPay\Platform_Checkout\Platform_Checkout_Utilities;
+
+
+/**
+ * WC_Payments_Checkout
+ */
+class WC_Payments_UPE_Checkout extends WC_Payments_Checkout {
+
+	/**
+	 * WC Payments Gateway.
+	 *
+	 * @var UPE_Payment_Gateway
+	 */
+	protected $gateway;
+
+	/**
+	 * Platform Checkout Utilities.
+	 *
+	 * @var Platform_Checkout_Utilities
+	 */
+	protected $platform_checkout_util;
+
+	/**
+	 * WC Payments Account.
+	 *
+	 * @var WC_Payments_Account
+	 */
+	protected $account;
+
+	/**
+	 * WC Payments Customer Service
+	 *
+	 * @var WC_Payments_Customer_Service
+	 */
+	protected $customer_service;
+
+	/**
+	 * List of the available payment gateways.
+	 *
+	 * @var array $gateway_map
+	 */
+	protected $gateway_map = [];
+
+	/**
+	 * Construct.
+	 *
+	 * @param UPE_Payment_Gateway          $gateway                WC Payment Gateway.
+	 * @param Platform_Checkout_Utilities  $platform_checkout_util Platform Checkout Utilities.
+	 * @param WC_Payments_Account          $account                WC Payments Account.
+	 * @param WC_Payments_Customer_Service $customer_service       WC Payments Customer Service.
+	 * @param array                        $gateway_map            WC Payment Gateway list.
+	 */
+	public function __construct(
+		UPE_Payment_Gateway $gateway,
+		Platform_Checkout_Utilities $platform_checkout_util,
+		WC_Payments_Account $account,
+		WC_Payments_Customer_Service $customer_service,
+		array $gateway_map
+	) {
+		$this->gateway                = $gateway;
+		$this->platform_checkout_util = $platform_checkout_util;
+		$this->account                = $account;
+		$this->customer_service       = $customer_service;
+		$this->gateway_map            = $gateway_map;
+
+		add_action( 'wc_payments_set_gateway', [ $this, 'set_gateway' ] );
```

Comment:
I was wondering whether we even need to do this at all. The only gateway-specific function that might get called in this class appears to be [this one](https://github.com/Automattic/woocommerce-payments/blob/merge-develop-into-poc/upe-instances-multiplied/includes/class-wc-payments-upe-checkout.php#L123). This is an edgy-casey function that is only called on the Order Pay page.

I'm not sure if there's a better way to handle this, but I suppose we can leave it this way for now and resolve it in another refactor sometime later.

Reply:
Currently, when not setting the gateway and using the card gateway, payment gateways are not rendered on the UI for the classic UPE checkout. I agree that we could resolve it in another refactor to reduce the number of things updated simultaneously, which increases the probability of unintentionally breaking other areas.

Reply:
> Currently, when not setting the gateway and using the card gateway, payment gateways are not rendered on the UI for the classic UPE checkout.

Cool, let's leave it and not break everything for now then. 😁 

> and using the card gateway

That's weird though. The card gateway shouldn't be using this code at all. Anyways, no matter now: let's just keep things working and moving forward.